decode_morse.rb

```diff
@@ -0,0 +1,54 @@
+@morse_chat = {
+  '.-' => 'A',
+  '-...' => 'B',
+  '-.-.' => 'C',
+  '-..' => 'D',
+  '.' => 'E',
+  '..-.' => 'F',
+  '--.' => 'G',
+  '....' => 'H',
+  '..' => 'I',
+  '.---' => 'J',
+  '-.-' => 'K',
+  '.-..' => 'L',
+  '--' => 'M',
+  '-.' => 'N',
+  '---' => 'O',
+  '.--.' => 'P',
+  '--.-' => 'Q',
+  '.-.' => 'R',
+  '...' => 'S',
+  '-' => 'T',
+  '..-' => 'U',
+  '...-' => 'V',
+  '.--' => 'W',
+  '-..-' => 'X',
+  '-.--' => 'Y',
+  '--..' => 'Z'
+}
+
+def decode_char(char)
+  @current_char = @morse_chat[char]
+  @current_char
+end
+
+def decode_word(word)
+  @current_word = word.split
+  @decoded_word = ''
+  @current_word.each do |char|
+    decode_char(char)
+    @decoded_word += decode_char(char)
+  end
+  print "#{@decoded_word} "
+end
+
+def decode_message(message)
+  @words = ''
+  @words = message.split('   ')
+  @words.each do |word|
+    decode_word(word)
+  end
+  @words
+end
```

Comment:
- [ ] There seem to be some discrepancies with respect to your functions. I am trying to print messages but I keep getting some irregularities. Your **`decode_word`** method should return the `@decode_word`, variable not print it, so that the **`decode_message`** method can use it.

- [ ] Also, as you can see the **`decode_message`** does not also return the decoded message, just an array of what was passed into it.

  ![vudej-decode](https://user-images.githubusercontent.com/36542986/160842604-5c3f7006-9854-45c0-ac93-6331a32ba454.jpg)

  **SUGGESTION:** You can refactor your methods to be something like so:
  
  **`decode_word`** method:
  ```
  def decode_word(word)
    @current_word = word.split
    @decoded_word = ''
    @current_word.each do |char|
      decode_char(char)
      @decoded_word += decode_char(char)
    end
    
    @decoded_word
  end
  ```

  **`decode_message`** method:
  ```
  def decode_message(message)
    @words = message.split('   ')
    @decoded_msg = ''
  
    @words.each do |word|
      @decoded_msg += "#{decode_word(word)} "
    end
    
    @decoded_msg.strip
  end
  ```